### PR TITLE
Packaging: first cut of working deb and rpm build

### PR DESCRIPTION
**Pull request by ceocoder**
* to build deb & rpm simply run
            ```
            version=<se.mv.er> vagrant provision
            ```
          vagrant shell provisioner will run the inline script and generate
          a deb and an rpm file in the local directory. This is achieved by
          multivm Vagrantfile one for centos and one for debian
        * To install the generated deb - run sudo dpkg -i
        * /pgcli/pgcli_<se.mv.er>.deb (then sudo apt-get install -f if deps
          are missing on a clean debian install)
        * To install generated rpm - sudo yum install path-to-rpm
        * package will be installed under /usr/share/pgcli
        * script pgcli will be a symlink in /usr/local/bin/pgcli ->
          /usr/share/pgcli/bin/pgcli (with right shebang line
          #!/usr/share/pgcli/bin/python)

### CODE REVIEW

**Comment by ceocoder** [comment]
@amjith added another provisioner for CentOS

**Comment by amjith** [comment]
I'm getting the following error:

```
There are errors in the configuration of this machine. Please fix
the following errors and try again:

vm:
* The box 'chef/debian-7.8' could not be found.
```

I presume that I need to download a vagrant box from here: http://www.vagrantbox.es/

I don't know which one to download. Can you help?

**Comment by ceocoder** [comment]
Hey I think you'll have to update your version of vagrant - the box I've referenced is from [atlas](https://atlas.hashicorp.com/boxes/search) Hashicorp's vagrant repo

**Comment by amjith** [comment]
I upgraded vagrant and tried again. New errors:

```
==> debian: Installing collected packages: click, Pygments, jedi, docopt, six, wcwidth, prompt-toolkit, psycopg2, sqlparse, pgcli
==> debian:   Running setup.py install for jedi
==> debian:   Running setup.py install for docopt
==> debian:   Running setup.py install for prompt-toolkit
==> debian:   Running setup.py install for psycopg2
==> debian:   Running setup.py install for sqlparse
==> debian:   Running setup.py install for pgcli
==> debian: Successfully installed Pygments-2.0.2 click-4.0 docopt-0.6.2 jedi-0.8.1 pgcli-0.16.3 prompt-toolkit-0.26 psycopg2-2.6 six-1.9.0 sqlparse-0.1.14 wcwidth-0.1.4
==> debian: -> Cleaning Virtualenv
==> debian: -> Removing compiled files
==> debian: -> Creating PgCLI deb
==> debian: {:message=>"All flags should be before the first argument (stray flags found: -d-d-p--after-install--after-remove--url--description--license", :level=>:warn, :timestamp=>"2015-04-15T04:36:04.   %6N+0000"}
==> debian: {:message=>"Invalid package configuration: Cannot package the path 'build/all', does it exist?", :level=>:error, :timestamp=>"2015-04-15T04:36:04.   %6N+0000"}
```

**Comment by amjith** [comment]
I may have forgotten to add the version number. I'm trying it again.

**Comment by amjith** [comment]
Cool. The deb package was created successfully. 

But rpm creation didn't start. 

```
==> centos: VM not created. Moving on...
```

I'm fairly new to vagrant, so I'm still learning by fire. If you have pointers on how to get the centos image going, please let me know.

**Comment by amjith** [comment]
Woot! Woot! It works. I'll update the develop.rst to document the steps to get it working.

Thank you! :tangerine:

**Comment by ceocoder** [comment]
@amjith can you try,
```bash
version=1.2.3 vagrant up centos
```

**Comment by amjith** [comment]
@ceocoder I managed to get both rpm and deb packages building. I've documented the steps in DEVELOP.rst file. 

https://github.com/amjith/pgcli/commit/e1958c0f9bad34edd6cb9775a28c276655729505#diff-e499c009307a0b632d20c2ecaef3a127

Can you please check if those instructions look good? 

Thank you very much for creating the Vagrantfile. It was a great exercise for me to learn how to work with it.

**Comment by ceocoder** [comment]
copied from twttr -  

I left a TODO in the license part https://github.com/amjith/pgcli/commit/b8d52953820085eeba47bfde3b51b89a68bc0f8f#diff-23b6f443c01ea2efcb4f36eedfea9089R88 … wasn't sure which license you were using

**Comment by ceocoder** [comment]
instructions make sense - `version=x.y.z` part is somewhat ambiguous, that allows someone to create a deb/rpm with different version number than what you have in `setup.py` - I can look that up and use it by default or add this to a `release.py`

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    sudo apt-get install -y libpq-dev python-dev python-setuptools rubygems
+    sudo easy_install pip
+    sudo pip install virtualenv virtualenv-tools
+    sudo gem install fpm
+    echo "-> Cleaning up old workspace"
+    rm -rf build
+    mkdir -p build/usr/share
+    virtualenv build/usr/share/pgcli
+    build/usr/share/pgcli/bin/pip install -U pip distribute
+    build/usr/share/pgcli/bin/pip uninstall -y distribute
```
Comment (amjith) [inline Vagrantfile]: @ceocoder Can you please help me understand why you pip install distribute and then immediately uninstall distribute?

**Pull request merged**